File: Miscellaneous_SoleKeyEncryption.pl
#!/usr/bin/perl;

use strict;
use warnings;

use SOLEKEYENCRYPTION;

my @message = split(//, "merry xmas");
my @key = split(//, "A5FB17C4D8"); # you might want To avoid zeroes
my $go_secure = SOLEKEYENCRYPTION->new();

my $encrypted = $go_secure->encodeWord(\@message, \@key);
print ("\nMessage is '", join("", @message), "';\nEncrypted version is ", join(", ", @{$encrypted}));

my $decrypted = $go_secure->decodeWord($encrypted, \@key);
print("\n\nDecrypted version is '", $decrypted, "'.");

print "\n\n";

File: Miscellaneous_SoleKeyEncryption.py
#!/usr/bin/python

from SoleKeyEncryption import ClosedEncryption

message = list("merry xmas")
key = list("A5FB17C4D8") # you might want To avoid zeroes
go_secure = ClosedEncryption()

encrypted = go_secure.encodeWord(message, key)
print("Message is '", ''.join(message), "';\nEncrypted version is ", encrypted)

decrypted = go_secure.decodeWord(encrypted, key)
print("\nDecrypted version is '", decrypted, "'.")
